src/Microsoft.Identity.Abstractions/DownstreamRestApi/DownstreamRestApiOptionsNoHttpMethod.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net.Http;
+
+namespace Microsoft.Identity.Abstractions
+{
+    /// <summary>
+    /// Specialization of <see cref="DownstreamRestApiOptions"/> where the <see cref="AuthorizationHeaderProviderOptions.HttpMethod"/>
+    /// is not settable beyond it's construction.
+    /// </summary>
+    /// <remarks>This class is useful to provide a better developer experience on the specialized methods
+    /// of <see cref="IDownstreamRestApi"/> where the HTTP method is provided already by the name of the 
+    /// method, and should not be overriden by the options.</remarks>
+    public class DownstreamRestApiOptionsNoHttpMethod : DownstreamRestApiOptions
```

Comment:
This is what you rightly proposed. The methods which start with a verb shouldn't let the possibility of overriding the Http method?

Reply:
```CSharp
           // Call a service based on the configuration only.
            downstreamRestApi.CallRestApiAsync("service");

            // Calls a service based on the programmatic description only.
            downstreamRestApi.CallRestApiAsync(null,
                options =>
                {
                    options.HttpMethod = HttpMethod.Get;
                    options.BaseUrl = "https://monApi.domain.com";
                    options.RelativePath = "api/values";
                });

            // In the following call, it's not possible to set the HttpMethod in the delegate, as it would no
            // make sense: it's already provided in the name of the method
            // The following code does not build (on purpose):
            // downstreamRestApi.DeleteForAppAsync("serviceName", "todo", options => { options.HttpMethod = HttpMethod.Put });
```